tree/dataframe/inc/ROOT/RDF/RFilter.hxx

```diff
@@ -72,9 +72,12 @@ public:
 
    RFilter(const RFilter &) = delete;
    RFilter &operator=(const RFilter &) = delete;
-   // must call Deregister here, before fPrevDataFrame is destroyed,
-   // otherwise if fPrevDataFrame is fLoopManager we get a use after delete
-   ~RFilter() { fLoopManager->Deregister(this); }
+   ~RFilter() {
+      // must Deregister objects from the RLoopManager here, before the fPrevDataFrame data member is destroyed:
+      // otherwise if fPrevDataFrame is the RLoopManager, it will be destroyed before the calls to Deregister happen.
```

Comment:
Can you think of a more robust ownership model (for a future PR)? This sounds like someone like me could easily break things a couple of years down the road...

Reply:
I totally agree, hence the verbose comment.

I can't think of another way at the moment unfortunately, but I noted it down.